### PR TITLE
[RFR][1LP] Cleanup vms handle error

### PR DESCRIPTION
**Pull request by mshriver**
Running locally with wrapanapi fixes.

### CODE REVIEW

File: requirements/frozen_docs.py3.txt [hunk 1]
```diff
@@ -35,8 +35,10 @@ fauxfactory==3.0.2
 flake8==3.5.0
 funcsigs==1.0.2
 future==0.16.0
+gevent==1.3.5
```
Comment (mshriver) [inline requirements/frozen_docs.py3.txt]: These are mocked in the docs build, but freeze_all is going to keep adding them, so FINE lets let it try to do a docs build with them. :)

File: scripts/cleanup_old_vms.py [hunk 2]
```diff
@@ -188,7 +193,7 @@ def scan_vm(provider_key, vm, delta, match_queue, scan_failure_queue):
         logger.info('%r: VM %r did not match age requirement', provider_key, vm.name)
 
 
-def delete_vm(provider_key, vm, age, result_queue):
```
Comment (psav) [inline scripts/cleanup_old_vms.py]: **REQUIRED:** These seem to be going against the new model of always passing in the objects, I know there must be a reason for this, can you explain it a little?

Comment (mshriver) [reply scripts/cleanup_old_vms.py]: Yep - the objects being passed into the function are done through the pool_manager, and have to be pickled. The wrapanapi VM objects aren't pickle-able.  I know I was hitting it on trying to get the namedtuple results, and changed them going into the function as well, but I'll double check that behavior.

Comment (mshriver) [reply scripts/cleanup_old_vms.py]: Part of the reasoning is that this script is running independent of any tests, and the VM could have been deleted by tests or users between scanning the provider, scanning the vm, and trying to delete.  So each time we call `get_vm` on it to fetch it, though this is secondary and we could have vm.exists traps.

**Comment by psav** [comment]
@mshriver and I discussed some of the patterns that we are not particularly happy with. We have devised a potential forward plan for this, but we are happy with what we have, on the proviso that we add an underscore to the bad pattern functions.

**Pull request merged**